Doc/glossary.rst

```diff
@@ -272,6 +272,12 @@ Glossary
       keys can be any object with :meth:`__hash__` and :meth:`__eq__` methods.
       Called a hash in Perl.
 
+   dictionary comprehension
+      A compact way to process all or part of the items in a sequence
```

Comment:
A list/set/dict comprehension processes the items in an iterable.
If the entry for listcomp says 'sequence', it should be changed.  I also do not like the 'all or part of' phrase.  The iteration runs to completion.  If one wants to not run the iteration to completion, one must use an explicit for loop.  Perhaps the 'or part' refers to 'if' clauses.  But that is a form of processing.  If the iterable yields volatile items, they are gone.